src/js/player.js

```diff
@@ -1292,7 +1292,13 @@ class Player extends Component {
    * @method play
    */
   play() {
-    this.techCall_('play');
+    // Only calls the tech's play if we already have a src loaded
+    if (this.player_.currentSrc()) {
+      this.techCall_('play');
+    } else {
+      this.autoplay(true);
```

Comment:
With respect to the play event firing when autoplay is set, player.src(...) actually handles updating the video source and fires the play event if autoplay is true. In my general testing of this on different browsers, this seems to work.